HIP/hip-514-missing-hts-precompile-functions.md

```diff
@@ -0,0 +1,245 @@
+| `bytes metadata`          | The unique metadata of the NFT                                                                |
+| `address spenderId`       | The account id specifying an account that has been granted spending permissions on this nft   |
+| `string ledgerId`         | The ID of the network ledger                                                                  |
+
+### Solidity Function Signatures
+
+The ABI signature and hashes for token management functions are as follows:
+
+| hash        | effective signature                               | return                    |
+| ------------|---------------------------------------------------|---------------------------|
+| `927da105`  | `allowance(address, address, address)`            | `(uint256)`               |
+| `e1f21c67`  | `approve(address, address, uint256)`              | `(uint256)`               |
+| `7336aaf0`  | `approveNFT(address, address, uint256)`           | `(uint256)`               |
+| `098f2366`  | `getApproved(address, uint256)`                   | `(uint256, address)`      |
+| `f49f40db`  | `isApprovedForAll(address, address, address)`     | `(uint256, bool)`         |
+| `367605ca`  | `setApprovalForAll(address, address, bool)`       | `(uint256)`               |
+| `46de0fb1`  | `isFrozen(address, address)`                      | `(bool)`                  |
+| `f2c31ff4`  | `isKyc(address, address)`                         | `(bool)`                  |
+| `f069f712`  | `deleteToken(address)`                            | `(bool)`                  |
+| `15c981cb`  | `getTokenAutoRenewInfo(address)`                  | `(address, uint32)`       |
+| `ae7611a0`  | `getTokenCustomFees(address)`                     | `(CustomFee[])`           |
+| `a7daa18d`  | `getTokenDefaultFreezeStatus(address)`            | `(bool)`                  |
+| `335e04c1`  | `getTokenDefaultKycStatus(address)`               | `(bool)`                  |
+| `3f28a19b`  | `getFungibleTokenInfo(address)`                   | `(FungibleTokenInfo)`     |
+| `1f69565f`  | `getTokenInfo(address)`                           | `(TokenInfo)`             |
+| `8946344a`  | `getTokenKey(address, uint)`                      | `(KeyValue)`              |
+| `2c20dcd1`  | `getNonFungibleTokenInfo(address, uint32)`        | `(NonFungibleTokenInfo)`  |
+| `5b8f8584`  | `freezeToken(address, address)`                   | `(bool)`                  |
+| `a9b86b17`  | `unFreezeToken(address, address)`                 | `(bool)`                  |
+| `8f8d7f99`  | `grantTokenKyc(address, address)`                 | `(bool)`                  |
+| `af99c633`  | `revokeTokenKyc(address, address)`                | `(bool)`                  |
+| `7c41ad2c`  | `pauseToken(address)`                             | `(bool)`                  |
+| `3f8cc5c2`  | `unPauseToken(address)`                           | `(bool)`                  |
```

Comment:
I think that for following functions: freezeToken, unFreezeToken, grantTokenKyc, revokeTokenKyc, pauseToken, unPauseToken would be better if the return type is (int) (the response code of the precompile execution) instead of (bool).

Reply:
Agreed having response codes is helpful, will add

Reply:
Updated